api/service/projects_service.go

```diff
@@ -91,15 +103,44 @@ func (service *projectsService) ListProjects(ctx context.Context, name string, u
 	return allProjects, nil
 }
 
-func (service *projectsService) UpdateProject(ctx context.Context, project *models.Project) (*models.Project, error) {
+func (service *projectsService) UpdateProject(ctx context.Context, project *models.Project) (*models.Project, string,
+	error) {
 	if service.authEnabled {
 		err := service.updateAuthorizationPolicy(ctx, project)
 		if err != nil {
-			return nil, fmt.Errorf("error while updating authorization policy for project %s", project.Name)
+			return nil, "", fmt.Errorf("error while updating authorization policy for project %s", project.Name)
+		}
+	}
+
+	updatedProject, err := service.save(project)
+	if err != nil {
+		return nil, "", fmt.Errorf("error saving project %s: %w", project.Name, err)
+	}
```

Comment:
Can we just return `err` directly? This maintains the behaviour of the existing implementation (i.e. we don't modify the err string returned from `service.save(project)`).